frank/utilities.py

```diff
@@ -64,78 +64,169 @@ class UVDataBinner(object):
 
     def __init__(self, uv, V, weights, bin_width):
         nbins = np.ceil(uv.max() / bin_width).astype('int')
+        #Protect against rounding
+        if nbins * bin_width < uv.max():
+            bins += 1
         bins = np.arange(nbins+1, dtype='float64') * bin_width
 
-        # Initialize binned data
-        bin_n = np.zeros(nbins, dtype='int64')
-        bin_uv = np.zeros(nbins, dtype='float64')
-        bin_wgt = np.zeros(nbins, dtype='float64')
+        self._bins = bins
+        self._nbins = nbins
+        self._norm =  1 / bin_width
```

Comment:
This line reminds me that we should force Python 3 as a requirement, both in the `setup.py` and in the documentation.
Python 2 has been discontinued and this is the major reason for which we should not support Python 2.

Reply:
I've opened an issue reminding us to do this